experimental/perf/scripts/test_case_helper.py

```diff
@@ -1,35 +1,97 @@
+
+    The goal is to measure the performance impact of runs with more steps.
+
+    Args:
+        root_folder (str): The root directory where results will be stored.
+        list_of_steps (list[int]): The list of steps used in the testing.
+    """
     loop = 1
+    mc = 100
+    sort_key = 1
+
     for step in list_of_steps:
-        log_folder = os.path.join(root_folder, f"loop{loop}_step{step}")
-        os.makedirs(log_folder, exist_ok=True)
-        capture_sar_metrics(log_folder)
-        run_experiment(loop, step, output_file=f"{log_folder}/results.json")
-        process_sar_files(log_folder, "metrics.json")
+        run_experiment_helper(loop, step, mc, root_folder, sort_key)
+        sort_key += 1
+
 
+def bench_log_scale_metric(root_folder: str, list_of_metric_count: list[int]) -> None:
+    """Runs a set of load tests with increasing # of metrics per step.
 
-def bench_log_scale_metric(root_folder: str, list_of_metric_count: list[int]):
+    The goal is to measure the performance impact of more metrics logged per step.
+
+    Args:
+        root_folder (str): The root directory where results will be stored.
+        list_of_metric_count (list[int]): The list of metric counts used in the testing.
+    """
     loop = 1
     step = 1000
+    sort_key = 1
     for mc in list_of_metric_count:
-        log_folder = os.path.join(root_folder, f"loop{loop}_step{step}_metriccount{mc}")
-        os.makedirs(log_folder, exist_ok=True)
-        capture_sar_metrics(log_folder)
-        run_experiment(loop, step, mc, output_file=f"{log_folder}/results.json")
-        process_sar_files(log_folder, "metrics.json")
+        run_experiment_helper(loop, step, mc, root_folder, sort_key)
+        sort_key += 1
+
+
+def run_experiment_helper(
+    loop: int,
+    step: int,
+    mc: int,
+    root_folder: str,
+    sort_key: int,
+    output_file="results.json",
+) -> None:
+    """A helper to do the standard perf test setup.
+
+    1) create a folder for this particular load test iteration
+    2) start capturing resource metrics
+    3) run the actual load tests
+    4) end the resource metrics and compute the summary stats
+
+    Args:
+        loop (int): The current loop iteration number.
+        step (int): Number of steps within the loop.
+        mc (int): Number of metrics to log per step.
+        root_folder (str): The root directory where results will be stored.
+        sort_key (str): A key used for sorting the test data (for naming the folder).
+        output_file (str, optional): The name of the file to store the results. Defaults to "results.json".
+
+    Returns:
+        None: This function does not return any value. It performs file and metric operations.
+
+    """
+    log_folder = Path(root_folder) / f"loop{loop}_step{step}_metriccount{mc}_{sort_key}"
+
+    log_folder.mkdir(parents=True, exist_ok=True)
+
+    capture_sar_metrics(log_folder)
+
+    run_experiment(loop, step, mc, output_file=f"{log_folder}/{output_file}")
```

Comment:
```suggestion
    run_experiment(loop, step, mc, output_file=str(log_folder/output_file))
```
This way you keep the correct path for each os